### PR TITLE
Sustainer IAP

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
In this PR we're setting up the Sustainer IAP Infrastructure. The core of this PR revolves around the `StoreManager` class, which encapsulates all the things IAP for us.

Ref. #1491

### Important!
The following pendings will be addressed in a follow up PR:

- Price Alert: Users will be able to pick between Yearly / Monthly subscription
- Tracks

I'm opting to keep this PR as leaner as possible, since I'm already exceeding the 500LOC ballpark.
**Thank you!!**

### Test
1. Edit the scheme and open `Options`
2. Make sure the `TestConfiguration.storekit` **StoreKit configuration** is selected
3. Launch Simplenote and log into your account
4. Open the Sidebar
5. Tap on the sustainer banner

- [ ] Verify that if you go thru the IAP flow, the Banner switches to `Sustainer`
- [ ] Verify that signing on in a different device gets you the same banner
- [ ] Verify that if you delete the subscription (Xcode / Store Debugging) and relaunch, the banner reverts

### Release
`RELEASE-NOTES.txt` was updated in f0e7aa6f with:
 
> Simplenote Users now have the ability to become sustainers #1493

### CODE REVIEW

File: Simplenote/Classes/SPSettingsViewController+Extensions.swift [hunk 1]
```diff
@@ -1,16 +1,36 @@
 import UIKit
 
 
-// MARK: - Interface Helpers
+// MARK: - Subscriber UI
 //
 extension SPSettingsViewController {
 
+    private var isActiveSustainer: Bool {
+        SPAppDelegate.shared().simperium.preferencesObject().isActiveSubscriber
```
Comment (jleandroperez) [inline Simplenote/Classes/SPSettingsViewController+Extensions.swift:9]: The `preferencesObject` getter ensures that if such object wasn't available, it'll be inserted. This is expected to default to `false`

File: Simplenote/SPPrivacyViewController.swift [hunk 2]
```diff
@@ -27,11 +27,11 @@ class SPPrivacyViewController: SPTableViewController {
     ///
     private var isAnalyticsEnabled: Bool {
         let simperium = SPAppDelegate.shared().simperium
-        guard let preferences = simperium.preferencesObject() else {
+        guard let isAnalyticsEnabled = simperium.preferencesObject().analytics_enabled else {
             return true
         }
 
-        return preferences.analytics_enabled?.boolValue ==  true
+        return isAnalyticsEnabled.boolValue
```
Comment (jleandroperez) [inline Simplenote/SPPrivacyViewController.swift:34]: I was in the neighborhood andd... had to fix this 😄

**Review by bjhomer — APPROVED** [review]
This all looks good to me!

**Comment by jleandroperez** [comment]
Thank you BJ!!

**Pull request merged**